pkg/util/http/transport.go

```diff
@@ -21,13 +21,28 @@ import (
 )
 
 var (
-	// NoProxyWarningMap map containing URL's whos proxy behavior will change in the future.
-	NoProxyWarningMap = make(map[string]bool)
+	// NoProxyIgnoredWarningMap map containing URL's who will ignore the proxy in the future
+	NoProxyIgnoredWarningMap = make(map[string]bool)
 
-	// NoProxyWarningMapMutex Lock for NoProxyWarningMap
+	// NoProxyUsedInFuture map containing URL's that will use a proxy in the future
+	NoProxyUsedInFuture = make(map[string]bool)
+
+	// NoProxyChanged map containing URL's whos proxy behavior will change in the future
+	NoProxyChanged = make(map[string]bool)
+
+	// NoProxyWarningMapMutex Lock for NoProxyIgnoredWarningMap
```

Comment:
this mutex is now used for all the `NoProxy` maps defined above, so let's update the comment + update its name to a more generic name that reflects that